src/audio/dcblock/dcblock.c

```diff
@@ -0,0 +1,530 @@
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/dcblock/dcblock.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/memory.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+static void dcblock_free_config(struct sof_dcblock_config **config)
+{
+	rfree(*config);
+	*config = NULL;
+}
+
+static int dcblock_set_coeffs(struct comp_data *cd)
+{
+	int i;
+
+	/* Verify that the configuration values are set */
+	if (!cd->config)
+		return -EINVAL;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++)
+		cd->dcblock[i].R_coeff = cd->config->R_coeffs[i];
+
+	return 0;
+}
+
+/*
+ * Sets the DC Blocking filter in pass through mode.
+ * The frequency response of a DCB filter is:
+ * H(z) = (1 - z^-1)/(1-Rz^-1).
+ * Setting R to 1 makes the filter acts as a passthrough component.
+ */
+static int dcblock_set_passthrough(struct comp_data *cd)
+{
+	comp_cl_info(&comp_dcblock, "dcblock_set_passthrough()");
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++)
+		cd->dcblock[i].R_coeff = Q_CONVERT_FLOAT(1, 31);
```

Comment:
Is R_coeff int32_t? Maybe it would be better to use Q2.30 to get exactly 1.0 value since 32 bit Q1.31 can hold only max 0.9999... I'm not sure what the macro produces when the range is exceeded.

Reply:
Yep, it's int32_t so this is a mistake. You have use here sat_int32() but the transfer function depending on computing precision for filter might not be right. It might be, maybe you have experience? I think it would be simple to switch to Q2.30 coefficient unless there's legacy with configuration tool.